packages/SwingSet/src/kernel/kernel.js

```diff
@@ -390,14 +396,15 @@ export default function buildKernel(
 
   async function deliverToVat(vatID, target, msg) {
     insistMessage(msg);
-    const vat = ephemeral.vats.get(vatID);
     kernelKeeper.incStat('dispatches');
     kernelKeeper.incStat('dispatchDeliver');
-    if (!vat) {
+    // eslint-disable-next-line no-use-before-define
+    if (!vatWarehouse.lookup(vatID)) {
       resolveToError(msg.result, VAT_TERMINATION_ERROR);
     } else {
       const kd = harden(['message', target, msg]);
-      const vd = vat.translators.kernelDeliveryToVatDelivery(kd);
+      // eslint-disable-next-line no-use-before-define
+      const vd = vatWarehouse.kernelDeliveryToVatDelivery(vatID, kd);
```

Comment:
let's make a note, we can simplify this (remove this method) by making VatWarehouse responsible for the `kernelSlog.delivery()` call (which wants both the kernelDelivery and the vatDelivery objects).. no need to expose the translation interface just for the sake of logging if we can move the logging closer to the translator's new home

Reply:
oh, well, ok we'll have to refactor `processNotify`, which needs to translate first, then delete the clist entries, *then* deliver.. maybe move that whole process into VatWarehouse

I've got a similar function coming down the pipe for GC operations, with the same sort of ordering requirements. Maybe we should move some of these `processXYZ` functions into their own file, and have the VatWarehouse import them and call them itself, rather than `kernel.js` doing it the other way around.